themes/default/content/blog/fullstack-pulumi-mern-stack-digitalocean/index.md

```diff
@@ -0,0 +1,680 @@
+
+When you deploy this app in a moment with [`pulumi up`]({{< relref "/docs/reference/cli/pulumi_up" >}}), Pulumi will provision a new MongoDB cluster (which usually takes a few minutes), and then once that's available, DigitalOcean will take our spec and use it to fetch the components of the app from GitHub at the specified branch and build them. From that point forward, any commit you make to that branch will trigger DigitalOcean to fetch, rebuild, and redeploy the app automatically.
+
+Make sure you've installed the DigitalOcean GitHub app as described above --- you should see it listed at <https://github.com/settings/installations>:
+
+![The Applications tab on GitHub, showing the DigitalOcean app installed](./digitalocean-github-app.png)
+
+Now return to the command line and run `pulumi up`:
+
+```bash
+$ pulumi up
+
+Previewing update (dev)
+
+View Live: https://app.pulumi.com/cnunciato/grocery-list/dev/previews/605bf32a-95b1-4221-bc35-0e667b30f38a
+
+     Type                                    Name              Plan
+ +   pulumi:pulumi:Stack                     grocery-list-dev  create
+ +   ├─ digitalocean:index:DatabaseCluster   cluster           create
+ +   ├─ digitalocean:index:DatabaseDb        db                create
+ +   ├─ digitalocean:index:App               app               create
+ +   └─ digitalocean:index:DatabaseFirewall  trusted-source    create
+
+Resources:
+    + 5 to create
+
+Do you want to perform this update?
+> yes
+
+Updating (dev)
+
+View Live: https://app.pulumi.com/cnunciato/grocery-list/dev/updates/1
+
+     Type                                    Name              Status
+     pulumi:pulumi:Stack                     grocery-list-dev
+ +   ├─ digitalocean:index:App               app               created
+ +   └─ digitalocean:index:DatabaseFirewall  trusted-source    created
+
+Outputs:
+  + liveUrl: "https://grocery-list-fb9bd.ondigitalocean.app"
+
+Resources:
+    + 2 created
+    3 unchanged
+
+Duration: 2m30s
+```
+
+Again, it'll probably take a few minutes to get everything spun up for the first time, but when the process completes, you'll have a working app at the URL provided by DigitalOcean, emitted as a Pulumi stack output:
+
+{{% choosable language typescript %}}
+
+```bash
+$ open $(pulumi stack output liveUrl)
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```bash
+$ open $(pulumi stack output live_url)
+```
+
+{{% /choosable %}}
+
+![The app now running in the DigitalOcean cloud](./deployed.png)
+
+You should also be able to explore your shiny new grocery-list app in the DigitalOcean Console, with all three of its components (and their build-time and runtime logs!) now represented:
+
+![The grocery-list app and its components in the DigitalOcean console](./in-the-console.png)
+
+Now try making a commit to your repository (any commit will do), and watch as the app redeploys automatically:
```

Comment:
I might be clear that we want them to make a change to the app rather than to infra here, right?

Reply:
Yep, good suggestion. I'll adjust for that.